kaolin/datasets/shapenet.py

```diff
@@ -193,6 +160,85 @@ def __getitem__(self, index):
         return {'data': data, 'attributes': attributes}
 
 
+class ShapeNet(KaolinDataset):
+    r"""ShapeNet Dataset class for meshes.
+    Args:
+        root (str): path to ShapeNet root directory
+        categories (list): List of categories to load from ShapeNet. This list may
+                           contain synset ids, class label names (for ShapeNetCore classes),
+                           or a combination of both.
+        train (bool): is using training split (default: True)
+        split (float): proportion of the dataset to split to the training datase
+    Returns:
+        .. code-block::
+        dict: {
+            attributes: {name: str, path: str, synset: str, label: str},
+            data: {vertices: torch.Tensor, faces: torch.Tensor}
+        }
+    Example:
+        >>> meshes = ShapeNet(root='../data/ShapeNet/')
+        >>> obj = meshes[0]
+        >>> obj['data'].vertices.shape
+        torch.Size([2133, 3])
+        >>> obj['data'].faces.shape
+        torch.Size([1910, 3])
+    """
+
+    def initialize(self, root: str, categories: list, train: bool = True, split: float = .7):
+        """Initialize the dataset
+           Args:
+               root (str): path to ShapeNet root directory
+               categories (list): List of categories to load from ShapeNet. This list may
+                                  contain synset ids, class label names (for ShapeNetCore classes),
+                                  or a combination of both.
+               train (bool): is using training split (default: True)
+               split (float): proportion of the dataset to split to the training dataset"""
+        self.root = Path(root)
+        self.paths = []
+        self.synset_idxs = []
+        self.synsets = _convert_categories(categories)
+        self.labels = [synset_to_label[s] for s in self.synsets]
+
+        # loops through desired classes
+        for i in range(len(self.synsets)):
+            syn = self.synsets[i]
+            class_target = self.root / syn
+            if not class_target.exists():
+                raise ValueError('Class {0} ({1}) was not found at location {2}.'.format(
+                    syn, self.labels[i], str(class_target)))
+
+            # find all objects in the class
+            models = sorted(class_target.glob('*'))
+            stop = int(len(models) * split)
+            if train:
+                models = models[:stop]
+            else:
+                models = models[stop:]
+            self.paths += models
+            self.synset_idxs += [i] * len(models)
+
+        self.names = [p.name for p in self.paths]
+
+    def __len__(self):
+        """Returns the length of the dataset. """
+        return len(self.paths)
+
+    def _get_data(self, index):
+        synset_idx = self.synset_idxs[index]
+        obj_location = self.paths[index] / 'model.obj'
```

Comment:
Actually, we can support `ShapeNetCoreV1` as well as `ShapeNetCoreV2` if we check for both `model.obj` and `model_normalized.obj`. (V2 has a few categories renamed to `model_normalized.obj`). Can be done in another PR if you prefer.

Reply:
The PR is actually not about ShapeNet, the class is more here to test the base dataset. Let's set it for ShapeNetV1

Reply:
Agreed